.github/generate-quest-list.mjs

```diff
@@ -0,0 +1,331 @@
+ * @returns {string} The Android String Resource name of the quest's title.
+ */
+function getQuestTitleStringNames(questName, questFileContent) {
+    let stringResourceNames = questFileContent.match(/(?<=R\.string\.)quest_\w+/g);
+
+    if (stringResourceNames.length === 0) {
+        throw new Error(`Could not find the title string reference for quest '${questName}'.`);
+    }
+
+    if (stringResourceNames.length === 1) {
+        return stringResourceNames;
+    }
+
+    return stringResourceNames.filter(name => name.includes('title'));
+}
+
+
+/**
+ * @returns {Promise<string>} The quest list wiki page content as a string.
+ */
+async function getWikiTableContent() {
+    const apiUrl = new URL('https://wiki.openstreetmap.org/w/api.php');
+    apiUrl.searchParams.set('action', 'parse');
+    apiUrl.searchParams.set('format', 'json');
+    apiUrl.searchParams.set('prop', 'wikitext');
+    apiUrl.searchParams.set('formatversion', '2');
+    apiUrl.searchParams.set('page', 'StreetComplete/Quests');
+    apiUrl.searchParams.set('section', '1'); // "Released quest types" section
+
+    return new Promise((resolve, reject) => {
+        https.get(apiUrl, response => {
+            let data = '';
+
+            response.on('data', chunk => {
+                data += chunk;
+            });
+
+            response.on('end', () => {
+                resolve(JSON.parse(data).parse.wikitext);
+            });
+        }).on('error', error => {
+            reject(error);
+        });
+    });
+}
+
+
+/**
+ * @param {string} wikiPageContent
+ * @returns {WikiTableQuest[]}
+ */
+function parseWikiTable(wikiPageContent) {
+    const tableRows = wikiPageContent.split('|-');
+
+    tableRows.shift(); // Drop table header and everything before the table
+    tableRows.push(tableRows.pop().split('|}')[0]); // Drop everything after the table
+
+    const cells = [];
+
+    for (const [rowIndex, row] of tableRows.entries()) {
+        const rowCells = row.split('\n|').slice(1);
+
+        if (rowIndex > 0) {
+            const previousRowCells = cells[rowIndex - 1];
+            for (const [cellIndex, cell] of previousRowCells.entries()) {
+                if (cell.startsWith(wikiRowSpan2)) {
+                    rowCells.splice(cellIndex, 0, cell.slice(wikiRowSpan2.length));
+                }
+            }
+        }
+
+        cells.push(rowCells);
+    }
+
+    return cells.map((rowCells, rowIndex) => new WikiTableQuest(rowCells, rowIndex));
+}
+
+
+async function writeCsvFile() {
+    const newQuests = quests.filter(quest => quest.wikiOrder === -1);
+    const oldQuests = quests.filter(quest => quest.wikiOrder > -1);
+
+    const csvLines = [
+        '"Quest Name", "Question", "Default Priority", "Wiki Order"',
+        ...wikiTable.filter(quest => quest.isOutdated).map(quest => quest.csvString),
+        ',,,',
+        ...newQuests.map(quest => quest.csvString),
+        ',,,',
+        ...oldQuests.map(quest => quest.csvString),
+    ];
```

Comment:
Surely there's an npm module to do writing to a CSV properly?

Reply:
I tried to keep it dependency-free (apart from Node.js), as proper dependency management requires an extra `package.json` file and ongoing maintenance. Writing the CSV file manually may not as pretty, but I think it's good enough for a script like this.

Reply:
Yeah I guess you're safe until a quest gets a comma or something, although I guess Java may stop that happening.